Generators/include/Generators/Generator.h

```diff
@@ -0,0 +1,87 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See https://alice-o2.web.cern.ch/ for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \author R+Preghenella - August 2017
+
+#ifndef ALICEO2_EVENTGEN_GENERATOR_H_
+#define ALICEO2_EVENTGEN_GENERATOR_H_
+
+#include "FairGenerator.h"
+#include <vector>
+#include <array>
+
+namespace o2
+{
+namespace eventgen
+{
+
+  /*****************************************************************/
+  /*****************************************************************/
+    
+  class Generator : public FairGenerator
```

Comment:
could you add a few lines of docu what this class does (is supposed to provide). Is it a pure interface or does it have some base functionality? (Maybe then GeneratorBase or something would be better?)

Reply:
it is a pure interface between the FairPrimaryGenerator and a generic event generator, supposed to define a protocol to be implemented by the concrete realisations of the interface. I'd rather keep the name Generator, but as you like